src/main/java/seedu/address/model/transaction/TransactionList.java

```diff
@@ -89,6 +89,36 @@ public void setTransactions(List<T> transactions) {
         internalList.setAll(transactions);
     }
 
+    /**
+     * Retrieves a list of transactions filtered by the {@code date} they were added.
+     */
+    public TransactionList<T> getTransactionsOnDate(Date date) {
+        TransactionList<T> filteredTransactions = new TransactionList<T>();
+
+        for (T transaction : this) {
+            if (transaction.getDate().equals(date)) {
+                filteredTransactions.add(transaction);
+            }
+        }
+
+        return filteredTransactions;
+    }
+
+    /**
+     * Retrieves a list of expenses filtered by the month they were added, using {@code date}.
+     */
+    public TransactionList<T> getTransactionsInMonth(Date date) {
```

Comment:
I think these filter methods are fine for now but after we implement the 'find/filter' function can use the getFilteredList(Predicate) pattern to make it cleaner and more standardised